net/nimble/host/include/host/ble_store.h

```diff
@@ -131,6 +137,40 @@ union ble_store_value {
     struct ble_store_value_cccd cccd;
 };
 
+struct ble_store_status_event {
+    /**
+     * The type of object that failed to persist; one of the
+     * BLE_STORE_OBJ_TYPE_[...] codes.
+     */
+    int obj_type;
+
+    /**
+     * The type of event being reported; one of the BLE_STORE_EVENT_TYPE_[...]
+     * codes.
+     */
+    int event_code;
+
+    /**
+     * Additional data related to the event; the valid field is inferred from
+     * the obj_type,event_code pair.
+     */
+    union {
+        /**
+         * The record that failed to be written.  Valid for the following event
+         * types:
+         *     o BLE_STORE_EVENT_OVERFLOW
```

Comment:
Ok, please see below what I have in mind.

   union {
        /**
         * The record that failed to be written. Valid for all the obj_type and following event
         * types:
         *     o BLE_STORE_EVENT_OVERFLOW
         */
        const union ble_store_value *value;

        /**
         * The connection handle of the peer.  Valid for the following event
         * types:
         *     o BLE_STORE_EVENT_OVERFLOW_NEXT
         *
         *   and following obj_types:
         *  
         *     o BLE_STORE_OBJ_TYPE_OUR_SEC
         *     o BLE_STORE_OBJ_TYPE_PEER_SEC
         */
        uint16_t conn_handle;
    };

In this way, whenever we need to handle object which also uses BLE_STORE_EVENT_OVERFLOW_NEXT but  has different data than conn_handle, we can easly add it here.